include/swift/SIL/SILInstructionWorklist.h

```diff
@@ -33,8 +34,7 @@ namespace swift {
 
 /// Manages a list of instructions awaiting visitation.
 class SILInstructionWorklist final {
-  llvm::SmallVector<SILInstruction *, 256> worklist;
-  llvm::DenseMap<SILInstruction *, unsigned> worklistMap;
+  SmallBlotSetVector<SILInstruction *, 256> worklist;
```

Comment:
My only concern here is that the API is making a decision for the user here on what the size is here and also whether or not small vectors are used.

The way we generally work around this is by templating that parameter. A good example here is BlotSetVector. At the top we define a template for the storage and provide a sane default: https://github.com/apple/swift/blob/master/include/swift/Basic/BlotSetVector.h#L47.

Then we create a SmallBlotSetVector based off that template: https://github.com/apple/swift/blob/master/include/swift/Basic/BlotSetVector.h#L150.

Reply:
It's now templated over the vector and map used by the blot set vector.  The small variant takes just an unsigned template parameter and uses SmallVector and SmallDenseMap.